cobspec/src/main/java/com/saralein/cobspec/Main.java

```diff
@@ -0,0 +1,78 @@
+
+        if (validationErrors.isEmpty()) {
+            ArgsParser argsParser = new ArgsParser(args);
+            Integer port = argsParser.parsePort();
+            Path root = argsParser.parseRoot(home);
+            Routes routes = setupRoutes(root);
+            HttpServer.start(port, root, routes, logger);
+        } else {
+            logger.log(String.join("\n", validationErrors));
+        }
+    }
+
+    private static List<String> runValidationAndReturnErrors(String[] args, String home) {
+        List<Validator> validators = new ArrayList<Validator>(){{
+            add(new PortValidator());
+            add(new DirectoryValidator(home));
+        }};
+
+        return new ArgsValidation(validators).validate(args);
+    }
+
+    private static Routes setupRoutes(Path root) {
+        FileHelper fileHelper = new FileHelper(root);
+        FormStore formStore = new FormStore();
+        FormBody formBody = new FormBody();
+        FormModification formModification = new FormModification();
+
+        HashMap<Integer, String> errorMessages = new HashMap<Integer, String>(){{
+            put(404, "<center><h1>404</h1>Page not found.</center>");
+            put(405, "");
+        }};
+
+        return new RoutesBuilder(new DirectoryController(fileHelper),
+                                 new FileController(fileHelper),
+                                 new ClientErrorController(errorMessages))
+                    .addRoute("/redirect", Methods.GET.name(), new RedirectController())
+                    .addRoute("/form", Methods.GET.name(), new FormGetController(formStore, formBody))
+                    .addRoute("/form", Methods.POST.name(), new FormPostController(formStore, formBody, formModification))
+                    .addRoute("/form", Methods.PUT.name(), new FormPutController(formStore, formBody, formModification))
+                    .addRoute("/form", Methods.DELETE.name(), new FormDeleteController(formStore))
+                    .addRoute("/method_options", Methods.OPTIONS.name(), new OptionsController(Methods.allowNonDestructiveMethods()))
+                    .addRoute("/method_options", Methods.GET.name(), new DefaultController())
+                    .addRoute("/method_options", Methods.PUT.name(), new DefaultController())
+                    .addRoute("/method_options", Methods.POST.name(), new DefaultController())
+                    .addRoute("/method_options", Methods.HEAD.name(), new DefaultController())
+                    .addRoute("/method_options2", Methods.OPTIONS.name(), new OptionsController(Methods.allowGetAndOptions()))
+                    .addRoute("/method_options2", Methods.GET.name(), new DefaultController())
+                    .addRoute("/tea", Methods.GET.name(), new DefaultController())
+                    .addRoute("/coffee", Methods.GET.name(), new CoffeeController())
+                    .build();
+    }
+}
```

Comment:
Whoa, there seems to be a lot of domain knowledge required to configure and use this server. Is there a way to give it a much simpler interface? It might encourage more people to use if it's easy to use. Just spitballing here, so I might have the syntax way wrong but what if it were like

```
public static void Main(String[] args) {
  Logger logger = new Logger(new ConnectionLogger(System.Out));
  new Server(logger).configure((RouterBuilder router) -> {
    router
      .addStatic("/public");
      .get("/redirect", () -> new RedirectController());
      // ....
  }).start(args);
}
```